### PR TITLE
Phase2-hgx246 Read in flat file data using xml files for HGCal

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Read in flat file data from the xml files for HGCal

#### PR validation:

Tested using runTheMatrix test workflows 

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Geometry/HGCalCommonData/interface/HGCalGeomParameters.h [hunk 1]
```diff
@@ -118,9 +118,17 @@ class HGCalGeomParameters {
   void resetZero(std::vector<double>&);
 
   constexpr static double tan30deg_ = 0.5773502693;
+  constexpr static int philippeEE = 2;
+  constexpr static int philippeHE = 3;
+  constexpr static int katjaFile = 4;
```
Comment (cvuosalo) [inline Geometry/HGCalCommonData/interface/HGCalGeomParameters.h:123]: It's a bit odd to embed people's names into the source code. It means if you don't know Katja, then you won't understand this variable. I presume these names are temporary during HGCal design and will be replaced in the near future.

**Comment by cvuosalo** [comment]
+1

File: Geometry/HGCalCommonData/src/HGCalGeomParameters.cc [hunk 2]
```diff
@@ -1215,6 +1338,42 @@ void HGCalGeomParameters::loadSpecParsTrapezoid(const HGCalParameters& php) {
...
+    }
+    tileIndx_.clear();
+    tileType_.clear();
+    tileSiPM_.clear();
+    tileHEX1_.clear();
+    tileHEX2_.clear();
+    tileHEX3_.clear();
+    tileHEX4_.clear();
+    for (unsigned int k = 0; k < tileRMin_.size(); ++k) {
+      php.tileRingR_.emplace_back(std::make_pair(tileRMin_[k], tileRMax_[k]));
```
Comment (kpedro88) [inline Geometry/HGCalCommonData/src/HGCalGeomParameters.cc:1361]: don't need to use make_pair with emplace_back

File: Geometry/HGCalCommonData/src/HGCalGeomParameters.cc [hunk 3]
```diff
@@ -1215,6 +1338,42 @@ void HGCalGeomParameters::loadSpecParsTrapezoid(const HGCalParameters& php) {
...
+#ifdef EDM_ML_DEBUG
+      edm::LogVerbatim("HGCalGeom") << "TileRingR[" << k << "] " << tileRMin_[k] << ":" << tileRMax_[k];
+#endif
+    }
+    tileRMin_.clear();
+    tileRMax_.clear();
+    for (unsigned k = 0; k < tileRingMin_.size(); ++k) {
+      php.tileRingRange_.emplace_back(std::make_pair(tileRingMin_[k], tileRingMax_[k]));
```
Comment (kpedro88) [inline Geometry/HGCalCommonData/src/HGCalGeomParameters.cc:1369]: don't need to use make_pair with emplace_back

File: Geometry/HGCalCommonData/src/HGCalGeomParameters.cc [hunk 4]
```diff
@@ -1065,10 +1080,45 @@ void HGCalGeomParameters::loadSpecParsHexagon8(const cms::DDFilteredView& fv,
     }
   }
 
+  // Read in parameters from Philip's file
+  if (php.waferMaskMode_ > 1) {
```
Comment (kpedro88) [inline Geometry/HGCalCommonData/src/HGCalGeomParameters.cc:1084]: there are a lot of conditionals and comparisons here and in the similar constructs below.

does this noticeably increase the geometry initialization time?

File: Geometry/HGCalCommonData/src/HGCalGeomParameters.cc (hunk 3 above)
Comment (bsunanda) [reply Geometry/HGCalCommonData/src/HGCalGeomParameters.cc]: Done

File: Geometry/HGCalCommonData/src/HGCalGeomParameters.cc (hunk 2 above)
Comment (bsunanda) [reply Geometry/HGCalCommonData/src/HGCalGeomParameters.cc]: Done

File: Geometry/HGCalCommonData/src/HGCalGeomParameters.cc (hunk 4 above)
Comment (bsunanda) [reply Geometry/HGCalCommonData/src/HGCalGeomParameters.cc]: Changed

File: Geometry/HGCalCommonData/interface/HGCalGeomParameters.h (hunk 1 above)
Comment (bsunanda) [reply Geometry/HGCalCommonData/interface/HGCalGeomParameters.h]: Changed

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by cvuosalo** [comment]
+1

**Comment by bsunanda** [comment]
@kpedro88 Could you please approve this PR?

**Comment by kpedro88** [comment]
+upgrade

**Comment by silviodonato** [comment]
+1

**Pull request merged**